junit/src/main/java/cucumber/api/junit/Cucumber.java

```diff
@@ -91,6 +93,14 @@ protected Description describeChild(FeatureRunner child) {
     }
 
     @Override
+    public void filter(Filter filter) throws NoTestsRemainException {
+        // dont pass filter to children as ParentRunner does by default
+        if (!filter.shouldRun(getDescription())) {
+            children.clear();
+        }
```

Comment:
The `ParrentRunner` tracks a copy of the children and uses these in various methods. Simply clearing the children here is not satisfactory. Should the filter match the description the best soltuion would be to pass a filter that filters everything to super or otherwise pass the original filter.

This will seems ugly but this is an unfortunate side effect of trying to use categories in a manner they were not designed.

http://junit.org/junit4/javadoc/4.12/org/junit/experimental/categories/Categories.html
> Note that, for now, annotating suites with @Category has no effect. Categories must be annotated on the direct method or class.